CONTRIBUTING.txt

```diff
@@ -0,0 +1,99 @@
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an Administrator or Maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR).
+    2. Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias. If the Platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
+    3. A PR SHOULD be a minimal and accurate answer to exactly one identified and agreed-on problem.
+    4. A PR SHOULD refrain from adding additional dependencies to the FreeCAD project unless no other option is available.
+    5. Code submissions MUST adhere to the code style guidelines of the project if these are defined.
+    6. If a PR contains multiple commits, each commit MUST compile cleanly and add value to the history of the project. Checkpoint commits SHOULD be squashed.
+    7. A PR SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+    8. A PR MUST compile cleanly and pass project self-tests on all target Platforms.
+    9. Each commit message in a PR MUST succinctly explain what the commit achieves.
+    10. The PR Description MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
+    11. A “Valid PR” is one which satisfies the above requirements.
+
+6. Process
+
+    1. Change on the project follows the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+    2. To request changes, a User (end user, Contributor, Maintainer, or Administrator) logs an issue on the project Platform issue tracker.
+    3. The User or Contributor SHOULD write the issue by describing the problem they face or observe. Links to the forum or other resources are permitted but the issue SHOULD be complete and accurate and SHOULD NOT require the reader to visit the forum or any other Platform to understand what is being described.
+    4. Issue authors SHOULD strive to describe the minimum acceptable condition.
+    5. Issue authors SHOULD focus on User tasks and avoid comparisons to other software solutions.
+    6. The User or Contributor SHOULD seek consensus on the accuracy of their observation and the value of solving the problem.
+    7. To submit a solution to a problem, a Contributor SHALL create a pull request back to the project.
+    8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
+    9. To discuss a proposed solution, Users MAY comment on the Platform Pull Request. Forum conversations regarding the solution SHOULD be discouraged and conversation redirected to the Pull Request or the related issue.
```

Comment:
Don't really like it. Most users are not coders and have no interest in having a GH account. :/
Several complains having decision taken on limited community feedback in the forum, I guess it will just worsen the problem : feedback will be limited and biased.

Reply:
This seems quite different from the C4 original where comments/discussions are permitted "elsewhere".  It would be reasonable to say "discussions on technical aspects of a proposed solution" should take place on the platform. 
Discussions about the desired functionality of the solution will take place on the forum whether or not we have a policy against it.

Reply:
This PR discussion now has more than 175 comments.  Even though it's most of the same people as on the forum, it has remained civil and productive (THANK YOU!!)   The nature of the tool makes it easier to do this.  

I know there's going to be disagreement around this topic.  That's why the terms MAY and SHOULD are used here.  

My next statement might be controversial:  I believe we shouldn't cater to the lowest common denominator.  If someone feels that their opinion is important enough to be heard, it isn't that difficult to get a Github account so they can share it in an appropriate context.  We can't stop people from talking about issues on the forum or wherever they like but we also don't have to make it easy to fork the discussion and argue endlessly.

Also, this is going to disproportionately impact our forum moderators because it's going to require them to do the 'discouraging'.   

This one is important to me but I'm here to find consensus, not impose my will.